modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java

```diff
@@ -177,6 +179,30 @@ public final void collectMetrics(Request<?> request, Response<?> response) {
                 .map(List::size)
                 .orElse(0);
 
+            if (exceptionCount > 0) {
+                final List<Object> statusCodes = awsRequestMetrics.getProperty(AWSRequestMetrics.Field.StatusCode);
+                final long amountOfRequestRangeNotSatisfiedErrors = Objects.requireNonNullElse(statusCodes, List.of())
+                    .stream()
+                    .filter(e -> (Integer) e == REQUESTED_RANGE_NOT_SATISFIED.getStatus())
+                    .count();
+                if (amountOfRequestRangeNotSatisfiedErrors > 0) {
+                    // REQUESTED_RANGE_NOT_SATISFIED errors are expected errors due to RCO, so we would like not to count them along with
```

Comment:
According to [this comment](https://elasticco.atlassian.net/browse/ES-8636?focusedCommentId=139000) by @henningandersen , it may be preferable to count them among the rest of the client errors, but just filter them out later (e.g., in Kibana's Discover or dashboards). So I wonder if it's better to not decrease the exceptionCount, but still pass on the clientErrorAttributes in the incrementBy? Will this allow us to filter out 416s?

Reply:
IIUC, you are suggesting that we still maintain a single `es.repositories.exceptions.total` metric but with added attributes, e.g. error code, to faciliate differentiating them in reporting time. I think this could be a viable idea. It also defers the definition of client vs server errors at reporting time which seems like a good thing, i.e. extra flexibility. It feels one step further compared to [my earlier suggestion](https://github.com/elastic/elasticsearch/pull/109657#discussion_r1642111137).

In theory, each exception could have a different error status code. So we will need to record them separately here with this suggestion. But it seems worthwhile.

Reply:
I've adjusted the code to not to deduct the client error codes from the exception count. We just track client error codes separately with the error code as an attribute. I'm currently only tracking REQUESTED_RANGE_NOT_SATISFIED(416) errors, but happy to adjust to other client errors. 

Not sure if can track all errors, because errors like `TOO_MANY_REQUESTS` are also tracked in `statusCodes`.